src/HotChocolate/Caching/src/Caching/CacheControlConstraintsOptimizer.cs

```diff
@@ -29,29 +34,45 @@ public void OptimizeOperation(OperationOptimizerContext context)
 
         var constraints = ComputeCacheControlConstraints(context.CreateOperation());
 
-        if (constraints.MaxAge is not null)
+        if (constraints.MaxAge is not null || constraints.SharedMaxAge is not null)
         {
-            var cacheType = constraints.Scope switch
+            var headerValue = new StringBuilder(constraints.Scope switch
             {
                 CacheControlScope.Private => _cacheControlPrivateScope,
                 CacheControlScope.Public => _cacheControlPublicScope,
                 _ => throw ThrowHelper.UnexpectedCacheControlScopeValue(constraints.Scope)
-            };
-
-            var headerValue = string.Format(
-                _cacheControlValueTemplate,
-                cacheType,
-                constraints.MaxAge);
+            });
+            if (constraints.MaxAge is not null)
+            {
+                headerValue.Append(_cacheControlDelimiter);
+                headerValue.Append(_cacheControlMaxAge);
+                headerValue.Append('=');
+                headerValue.Append(constraints.MaxAge.Value);
+
+                context.ContextData.Add(
+                    WellKnownContextData.CacheControlConstraints,
+                    new ImmutableCacheConstraints(
```

Comment:
What is the purpose of this interface? It is public so I didn't dare changing it. Since MaxAge is non-nullable, it would break if I moved it outside of this if statement, as I would have to make MaxAge nullable.

Reply:
I will have a look at this